contextcache: assignment to nil map

just saw this in docker stack:
```
2016/01/13 03:03:24 [I] port present in settings
2016/01/13 03:03:24 [I] protocol present in settings
2016/01/13 03:03:24 [I] processing amqp message with routing key: INFO.monitor.created
panic: assignment to entry in nil map

goroutine 24 [running]:
github.com/grafana/grafana/pkg/api.(*ContextCache).AddMonitor(0xc82019a600, 0xc820124e00)
        /go/src/github.com/grafana/grafana/pkg/api/socketio.go:142 +0x516
github.com/grafana/grafana/pkg/api.EmitAddMonitor(0xc820124e00, 0x0, 0x0)
        /go/src/github.com/grafana/grafana/pkg/api/socketio.go:576 +0x78
github.com/grafana/grafana/pkg/api.eventConsumer(0xc8200ae500, 0x0, 0x0)
        /go/src/github.com/grafana/grafana/pkg/api/socketio.go:680 +0x610
github.com/grafana/grafana/pkg/services/rabbitmq.(*Consumer).consume.func1(0xc8205e2240, 0xc820469fa0, 0xc8201e6730)
        /go/src/github.com/grafana/grafana/pkg/services/rabbitmq/consumer.go:148 +0xe3
created by github.com/grafana/grafana/pkg/services/rabbitmq.(*Consumer).consume
        /go/src/github.com/grafana/grafana/pkg/services/rabbitmq/consumer.go:155 +0x11e
```

--- a/pkg/api/socketio.go
+++ b/pkg/api/socketio.go
@@ -106,7 +106,11 @@ func (c *ContextCache) loadMonitors() {
 		c.monitorsIndex = newIndex
 	}
 	for _, ctx := range c.Contexts {
-		ctx.MonitorsIndex = newCollectorIndex[ctx.Collector.Id]
+		cIdx, ok := newCollectorIndex[ctx.Collector.Id]
+		if !ok {
+			cIdx = make(map[int64]*m.MonitorDTO)
+		}
+		ctx.MonitorsIndex = cIdx
 	}
 }
 

